src/wasm.h

```diff
@@ -92,7 +92,21 @@ struct Name : public cashew::IString {
 typedef uint32_t Index;
 
 // An address in linear memory. For now only wasm32
-typedef uint32_t Address;
+struct Address {
+  typedef uint32_t address_t;
+  address_t addr;
+  Address() : addr(0) {}
+  Address(uint64_t a) : addr(static_cast<address_t>(a)) {
+    assert(a <= std::numeric_limits<address_t>::max());
+  }
+  Address& operator =(uint64_t a) {
```

Comment:
I like having spaces better in general, but consistency is better here I think. I'll change it.